extensions/wikidata/tests/src/org/openrefine/wikidata/qa/scrutinizers/DifferenceWithinScrutinizerTest.java

```diff
@@ -19,39 +45,67 @@ public EditScrutinizer getScrutinizer() {
+        List<Snak> snakList1 = Collections.singletonList(snak1);
+        List<Snak> snakList2 = Collections.singletonList(snak2);
+        List<Snak> snakList3 = Collections.singletonList(snak3);
+        SnakGroup snakGroup1 = Datamodel.makeSnakGroup(snakList1);
+        SnakGroup snakGroup2 = Datamodel.makeSnakGroup(snakList2);
+        SnakGroup snakGroup3 = Datamodel.makeSnakGroup(snakList3);
+        List<SnakGroup> snakGroupList = Arrays.asList(snakGroup1, snakGroup2, snakGroup3);
+        List<Statement> statementList = constraintParameterStatementList(entityIdValue, snakGroupList);
+
+        ConstraintFetcher fetcher = mock(ConstraintFetcher.class);
+        when(fetcher.getConstraintsByType(upperBoundPid, "Q21510854")).thenReturn(statementList);
+        when(fetcher.findValues(snakGroupList, "P2306")).thenReturn(Collections.singletonList(lowerBoundPid));
+        when(fetcher.findValues(snakGroupList, "P2313")).thenReturn(Collections.singletonList(minValue));
+        when(fetcher.findValues(snakGroupList, "P2312")).thenReturn(Collections.singletonList(maxValue));
+        setFetcher(fetcher);
+
         scrutinize(updateA);
         assertWarningsRaised(DifferenceWithinRangeScrutinizer.type);
     }
 
     @Test
     public void testNoIssue() {
         ItemIdValue idA = TestingData.existingId;
-        PropertyIdValue lowerBoundPid = MockConstraintFetcher.lowerBoundPid;
-        PropertyIdValue upperBoundPid = MockConstraintFetcher.differenceWithinRangePid;
-
         TimeValue lowerYear = new TimeValueImpl(2000, (byte)10, (byte)15, (byte)0, (byte)0, (byte)0, (byte)11, 0, 0, 0, TimeValue.CM_GREGORIAN_PRO);
         TimeValue upperYear = new TimeValueImpl(2020, (byte)10, (byte)15, (byte)0, (byte)0, (byte)0, (byte)11, 0, 0, 0, TimeValue.CM_GREGORIAN_PRO);
-
         ValueSnak value1 = Datamodel.makeValueSnak(lowerBoundPid, lowerYear);
         ValueSnak value2 = Datamodel.makeValueSnak(upperBoundPid, upperYear);
-
         Statement statement1 = new StatementImpl("P569", value1,idA);
         Statement statement2 = new StatementImpl("P570", value2,idA);
-
         ItemUpdate updateA = new ItemUpdateBuilder(idA).addStatement(statement1).addStatement(statement2).build();
+
+        Snak snak1 = Datamodel.makeValueSnak(propertyParameterPID, lowerBoundPid);
+        Snak snak2 = Datamodel.makeValueSnak(minimumValuePID, minValue);
+        Snak snak3 = Datamodel.makeValueSnak(maximumValuePID, maxValue);
+        List<Snak> snakList1 = Collections.singletonList(snak1);
+        List<Snak> snakList2 = Collections.singletonList(snak2);
+        List<Snak> snakList3 = Collections.singletonList(snak3);
+        SnakGroup snakGroup1 = Datamodel.makeSnakGroup(snakList1);
+        SnakGroup snakGroup2 = Datamodel.makeSnakGroup(snakList2);
+        SnakGroup snakGroup3 = Datamodel.makeSnakGroup(snakList3);
+        List<SnakGroup> snakGroupList = Arrays.asList(snakGroup1, snakGroup2, snakGroup3);
```

Comment:
The seven lines above could be simplified by introducing a function that takes a `List<Snak>` and returns a `List<SnakGroup>` by grouping the snaks with identical properties.